shell/pages/c/_cluster/longhorn/index.vue

```diff
@@ -19,18 +22,30 @@ export default {
     };
   },
 
-  computed: { ...mapGetters(['currentCluster']) },
+  computed: {
+    ...mapGetters(['currentCluster']),
 
-  mounted() {
-    this.externalLinks = [
-      {
-        enabled:     true,
-        iconSrc:     this.longhornImgSrc,
-        label:       'longhorn.overview.linkedList.longhorn.label',
-        description: 'longhorn.overview.linkedList.longhorn.description',
-        link:        `/k8s/clusters/${ this.currentCluster.id }/api/v1/namespaces/longhorn-system/services/http:longhorn-frontend:80/proxy/`
-      },
-    ];
+    uiService() {
+      return this.$store.getters['cluster/matching'](SERVICE, 'app=longhorn-ui')[0];
+    }
+  },
+
+  async mounted() {
+    if ( this.$store.getters['cluster/schemaFor'](SERVICE) ) {
+      await this.$store.dispatch('cluster/findAll', { type: SERVICE });
```

Comment:
There might be a large number of services, so for performance, we can use the label selector to look just for the longhorn one.

Reply:
That was my original approach with:
```js
  async mounted() {
    if ( this.$store.getters['cluster/schemaFor'](SERVICE) ) {
      const uiService = await this.$store.dispatch('cluster/findMatching', {
        type:     SERVICE,
        selector: 'app=longhorn-ui'
      });

      if ( !isEmpty(uiService) && uiService.metadata?.namespace ) {
        this.externalLinks = [
          {
            enabled:     true,
            iconSrc:     this.longhornImgSrc,
            label:       'longhorn.overview.linkedList.longhorn.label',
            description: 'longhorn.overview.linkedList.longhorn.description',
            link:        `/k8s/clusters/${ this.currentCluster.id }/api/v1/namespaces/${ uiService.metadata.namespace }/services/http:longhorn-frontend:80/proxy/`
          },
        ];
      }
    }
  },
```

The service is never found even when it exists. If I fetch all of the services with `findAll`, then run `findMatching` this will work but defeats the purpose. Am I using `findMatching` incorrectly?


https://github.com/rancher/dashboard/assets/40806497/1d605948-ab4a-4235-a423-c6e216ebe16f

Reply:
You can supply `force: true` in the options to bypass that behaviour.